pupil_src/shared_modules/camera_models.py

```diff
@@ -299,13 +219,121 @@ def solvePnP(
     def save(self, directory: str, custom_name: typing.Optional[str] = None):
         ...
 
+    subclass_by_cam_type = dict()
+
+    def __init_subclass__(cls, *args, **kwargs):
+        super().__init_subclass__(*args, **kwargs)
+        # register subclasses by cam_type
+        if cls.cam_type == ...:
+            raise NotImplementedError("Subclass needs to define 'cam_type'!")
+        if cls.cam_type in Camera_Model.subclass_by_cam_type:
+            raise ValueError(
+                f"Error trying to register camera model {cls}: Camera model with"
+                f" cam_type '{cls.cam_type}' already exists:"
+                f" {Camera_Model.subclass_by_cam_type[cls.cam_type]}"
+            )
```

Comment:
Is this really necessary? This prevents people from implementing their own class for a specific `cam_type`.

Reply:
I put it there since the DummyModel inherits from RadialModel and if you forget to specify cam_type when inheriting, your child class will take over everything from the parent class. This can easily happen and is most certainly not what you want.

Does it make sense to implement my own radial model with which I want to overwrite the behavior of the built-in model? I see my original reason as a more pressing issue.